subsys/fs/shell.c

```diff
@@ -334,6 +334,60 @@ static int cmd_read(const struct shell *shell, size_t argc, char **argv)
 	return 0;
 }
 
+static int cmd_cat(const struct shell *shell, size_t argc, char **argv)
+{
+	char path[MAX_PATH_LEN];
+	uint8_t buf[BUF_CNT];
+	struct fs_dirent dirent;
+	struct fs_file_t file;
+	int err;
+	ssize_t read;
+
+	fs_file_t_init(&file);
+
+	for (size_t i = 1; i < argc; ++i) {
+		create_abs_path(argv[i], path, sizeof(path));
+
+		err = fs_stat(path, &dirent);
+		if (err < 0) {
+			shell_error(shell, "Failed to obtain file %s (err: %d)",
+					path, err);
+			continue;
+		}
+
+		if (dirent.type != FS_DIR_ENTRY_FILE) {
+			shell_error(shell, "Note a file %s", path);
+			continue;
+		}
+
+		err = fs_open(&file, path, FS_O_READ);
+		if (err) {
```

Comment:
Sorry I missed that: `err < 0`. Generally when functions return -errno it is better to do `err < 0` to keep code more in line with processing error codes from functions that return values `>= 0` on success.

Reply:
I am sorry. I should fix that when I fixed the same for calling `stat`